locations/spiders/nissan_za.py

```diff
@@ -0,0 +1,46 @@
+import scrapy
+
+from locations.dict_parser import DictParser
+from locations.hours import DAYS, OpeningHours
+
+
+class NissanZASpider(scrapy.Spider):
```

Comment:
Suggest changing to `class NissanEuropeSpider(scrapy.Spider):`

It appears all the countries listed at "https://www.nissan-europe.com/" are using the same API and thus this spider can obtain dealer locations across dozens of countries.